### PR TITLE
share a measurement from the measurement tool

### PR DESCRIPTION
**Pull request by thiloSchlemmer**
enable the measureToolContent to publish a link with the current state of the measurement as:

- readonly link; the content of the measurement is not editable and is at the same state at any time
- writeable link; the content of the measurement is editable and can be updated by any user which have the shared link

### CODE REVIEW

**Review by taulinger — APPROVED** [review]
@thiloSchlemmer @alsturm Very nice,
would suggest just some changes to comments...
After that let's go for a merge!

**Comment by taulinger** [comment]
fixed the ba-button problem.

**Comment by taulinger** [comment]
https://github.com/ldbv-by/bav4-nomigration/blob/defd7a3c713f780ad0e440c09e92269397dd8b48/test/modules/toolbox/components/measureToolContent/MeasureToolContent.test.js#L148

Maybe we should also test alls ba-buttons for the expected value of the attribute "label"?

**Comment by thiloSchlemmer** [comment]
Similar to `it('displays the finish-button', async () => {...` but with detailed attribute-expectations?

**Comment by taulinger** [comment]
> Similar to:
> 
> https://github.com/ldbv-by/bav4-nomigration/blob/defd7a3c713f780ad0e440c09e92269397dd8b48/test/modules/toolbox/components/measureToolContent/MeasureToolContent.test.js#L93
> 
> ... but with detailed attribute-expectations?

 Something like
 `expect(element.getAttribute('label').toBe('toolbox_measureTool_delete_measure')`

File: src/modules/toolbox/components/measureToolContent/ShareMeasureDialog.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		const translate = (key) => this._translationService.translate(key);
+		const onCopyUrlToClipBoard = async () => this._copyValueToClipboard(url);
+
+		const getShareApiContent = (useShareApi) => {
+			if (useShareApi) {
+				const onClickWithApi = () => {
+					this._environmentService.getWindow().navigator.share({
+						title: translate('toolbox_measureTool_share_link_title'),
+						url: url
+					}).catch(console.error);
```
Comment (taulinger) [inline src/modules/toolbox/components/measureToolContent/ShareMeasureDialog.js:57]: `catch` does take a function as an argument.
 It also seems that the test is missing for that case

**Review by taulinger — COMMENTED** [review]
Almost 😉 done (see inline comment)

**Review by taulinger — APPROVED** [review]
🥳

**Pull request merged**